### ISSUE TITLE
V5: ClientEntityInterface, only one redirect URI?

### ISSUE THREAD

**Issue by kop** [thephpleague/oauth2-server#511]
Hi,

In `v4` it was possible to define multiple allowed redirect URIs for the client.

Do I understand correctly that it's not longer possible in `v5`:

```php
    /**
     * Set the client's redirect uri.
     *
     * @param string $redirectUri
     */
    public function setRedirectUri($redirectUri);

    /**
     * Returns the registered redirect URI.
     *
     * @return string
     */
    public function getRedirectUri();
```

**Comment by Bobselp** [comment]
I had the same issue. The way I solved it is that getRedirectUri() gets the requested redirect URI from the $_GET parameters, and validates it against the database backend. If it is valid, it simply returns the URI requested, if not it returns nothing. This works at least for the client credentials grant, I havn't tested it for the (two-step) auth code credentials grant yet.

**Comment by alexbilbie** [comment]
The getRedirectUri method on the client entity could return a string or an array of strings?

**Comment by alexbilbie** [comment]
This is fixed as of 8274c56
